atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/TimestampStateStoreTest.java

```diff
@@ -42,46 +54,116 @@ public void before() {
+        assertThat(timestampStateStore.getCommitInfo(100L))
+                .hasValue(TimestampStateStore.CommitInfo.of(WRITES_TOKEN, VERSION_2));
     }
 
     @Test
-    public void cannotPutCommitUpdateBeforeStartUpdate() {
-        TransactionUpdate update = TransactionUpdate.builder()
-                .startTs(100L)
-                .commitTs(400L)
-                .writesToken(LockToken.of(UUID.randomUUID()))
-                .build();
+    public void canPutMultipleUpdatesForDifferentStartTimestamps() {
+        timestampStateStore.putStartTimestamps(ImmutableSet.of(100L, 102L, 104L), VERSION_1);
+        timestampStateStore.putCommitUpdates(ImmutableSet.of(UPDATE, UPDATE_2), VERSION_2);
 
-        assertThatThrownBy(() -> timestampStateStore.putCommitUpdates(ImmutableSet.of(update), VERSION_2))
-                .isExactlyInstanceOf(TransactionLockWatchFailedException.class)
-                .hasMessage("start timestamp missing from map");
+        assertThat(timestampStateStore.getCommitInfo(100L))
+                .hasValue(TimestampStateStore.CommitInfo.of(WRITES_TOKEN, VERSION_2));
+        assertThat(timestampStateStore.getCommitInfo(102L))
+                .hasValue(TimestampStateStore.CommitInfo.of(WRITES_TOKEN_2, VERSION_2));
+    }
+
+    @Test
+    public void canRemoveVersions() {
+        timestampStateStore.putStartTimestamps(ImmutableSet.of(100L, 102L), VERSION_1);
+        timestampStateStore.putCommitUpdates(ImmutableSet.of(UPDATE), VERSION_2);
+
+        timestampStateStore.remove(100L);
+        assertThat(timestampStateStore.getStartVersion(100L)).isEmpty();
+        assertThat(timestampStateStore.getCommitInfo(100L)).isEmpty();
+        assertThat(timestampStateStore.getStartVersion(102L)).hasValue(VERSION_1);
+
+        timestampStateStore.remove(102L);
+        assertThat(timestampStateStore.getStartVersion(102L)).isEmpty();
+    }
+
+    @Test
+    public void canClear() {
+        timestampStateStore.putStartTimestamps(ImmutableSet.of(100L, 102L), VERSION_1);
+        timestampStateStore.putCommitUpdates(ImmutableSet.of(UPDATE), VERSION_2);
+
+        timestampStateStore.clear();
+        assertThat(timestampStateStore.getStartVersion(100L)).isEmpty();
+        assertThat(timestampStateStore.getCommitInfo(100L)).isEmpty();
+        assertThat(timestampStateStore.getStartVersion(102L)).isEmpty();
+        assertThat(timestampStateStore.getEarliestLiveSequence()).isEmpty();
+    }
+
+    @Test
+    public void earliestVersionUpdatesWhenAllTimestampsRemovedForVersion() {
+        timestampStateStore.putStartTimestamps(ImmutableSet.of(100L, 200L), VERSION_1);
+        timestampStateStore.putStartTimestamps(ImmutableSet.of(400L, 800L), VERSION_2);
+
+        assertThat(timestampStateStore.getEarliestLiveSequence()).hasValue(Sequence.of(1L));
+
+        removeAndCheckEarliestVersion(100L, 1L);
+        removeAndCheckEarliestVersion(800L, 1L);
+        removeAndCheckEarliestVersion(200L, 17L);
```

Comment:
I know this is a holdover, but these numbers are magical too. Can we just have this method take a version instead of a second long?